src/Symfony/Component/AssetMapper/Tests/Compiler/JavaScriptImportPathCompilerTest.php

```diff
@@ -277,6 +278,31 @@ public static function provideMissingImportModeTests(): iterable
         ];
     }
 
+    public function testErrorMessageAvoidsCircularException()
+    {
+        $assetMapper = $this->createMock(AssetMapperInterface::class);
+        $assetMapper->expects($this->any())
+            ->method('getAsset')
+            ->willReturnCallback(function ($logicalPath) {
+                if ('htmx' === $logicalPath) {
+                    return null;
+                }
+
+                if ('htmx.js' === $logicalPath) {
+                    throw new CircularAssetsException();
+                }
+            });
+
+        $asset = new MappedAsset('html.js', '/path/to/app.js');
```

Comment:
I think there's a typo here, shouldn't it be `'htmx.js'`?

Reply:
Yup - it doesn't matter for the test, but definitely wasn't my intention. Fixing it - thanks :)

Reply:
Well, with `html.js`, the test would not trigger a circular error, and so would not prevent regressions. So it definitely matters.

Reply:
No - the circular reference is triggered a bit manually on the mock. So you're correct in a real-world example, but in this case, this part doesn't matter (creating a test that resulted in a natural, real-world circular reference was going to be a lot more work as we need a real AssetMapper, MappedAssetFactory, etc.